keras/data/dataset.py

```diff
@@ -0,0 +1,11 @@
+# TODO add some default like hdf5 dataset, directory, etc
+class Dataset():
+    """
+    Base object for every dataset
+    """
+
+    def __getitem__(self, index):
+        raise NotImplementedError
+
```

Comment:
This is highly dataset-specific. I don't want to be forced to implement something that I may not need. If my dataset is already on disk, why should I implement something like this? This can still be done by the user. The same way that actual generators are used, there is no 'mandatory' thing to do.